editor/src/components/canvas/padding-utils.ts

```diff
@@ -0,0 +1,149 @@
+import { getLayoutProperty } from '../../core/layout/getLayoutProperty'
+import { MetadataUtils } from '../../core/model/element-metadata-utils'
+import { Either, isLeft, right } from '../../core/shared/either'
+import { ElementInstanceMetadataMap, isJSXElement } from '../../core/shared/element-template'
+import { CanvasVector } from '../../core/shared/math-utils'
+import { ElementPath } from '../../core/shared/project-file-types'
+import { assertNever } from '../../core/shared/utils'
+import { CSSNumber, CSSPadding } from '../inspector/common/css-utils'
+import { EdgePiece } from './canvas-types'
+
+type CSSPaddingKey = keyof CSSPadding
+export type SimpleCSSPadding = { [key in CSSPaddingKey]: number }
+
+export function simplePaddingFromMetadata(
+  metadata: ElementInstanceMetadataMap,
+  elementPath: ElementPath,
+): SimpleCSSPadding {
+  const element = MetadataUtils.findElementByElementPath(metadata, elementPath)
+
+  const defaultPadding: SimpleCSSPadding = {
+    paddingTop: 0,
+    paddingBottom: 0,
+    paddingLeft: 0,
+    paddingRight: 0,
+  }
+
+  if (element == null || isLeft(element.element) || !isJSXElement(element.element.value)) {
+    return defaultPadding
+  }
+
+  const padding = cssPaddingToSimple(
+    getLayoutProperty('padding', right(element.element.value.props), ['style']),
+    defaultPadding,
+  )
+
+  const paddingTop = pxValueFromEither(
+    getLayoutProperty('paddingTop', right(element.element.value.props), ['style']),
+  )
+
+  const paddingBottom = pxValueFromEither(
+    getLayoutProperty('paddingBottom', right(element.element.value.props), ['style']),
+  )
+
+  const paddingLeft = pxValueFromEither(
+    getLayoutProperty('paddingLeft', right(element.element.value.props), ['style']),
+  )
+
+  const paddingRight = pxValueFromEither(
+    getLayoutProperty('paddingRight', right(element.element.value.props), ['style']),
+  )
+
+  return cssPaddingWithDefaults(
+    { paddingTop, paddingBottom, paddingLeft, paddingRight },
+    padding,
+    defaultPadding,
+  )
+}
+
+function cssPaddingWithDefaults(
+  parts: Partial<SimpleCSSPadding>,
+  whole: Partial<SimpleCSSPadding>,
+  defaults: SimpleCSSPadding,
+): SimpleCSSPadding {
+  return {
+    paddingTop: parts.paddingTop ?? whole.paddingTop ?? defaults.paddingTop,
+    paddingBottom: parts.paddingBottom ?? whole.paddingBottom ?? defaults.paddingBottom,
+    paddingLeft: parts.paddingLeft ?? whole.paddingLeft ?? defaults.paddingLeft,
+    paddingRight: parts.paddingRight ?? whole.paddingRight ?? defaults.paddingRight,
+  }
+}
+
+const pxValue = (number: CSSNumber): number | undefined =>
+  number.unit === 'px' || number.unit == null ? number.value : undefined
+
+function pxValueFromEither(value: Either<string, CSSNumber | undefined>): number | undefined {
+  if (isLeft(value) || value.value == null) {
```

Comment:
If `pxValue` also returned an `Either` and handled a null input, you could flatMap it and use `defaultEither` to get the result.

Reply:
thanks! I've updated the code along your suggestion